src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx

```diff
@@ -137,26 +131,15 @@ function NodeDetailComponent({
                     )
                 }
 
-                if (Array.isArray(result.manifest.spec.ephemeralContainers)) {
-                    const ephemeralContainerStatusMap = new Map<string, string[]>()
-                    result.manifest.spec.ephemeralContainers.forEach((con) => {
-                        ephemeralContainerStatusMap.set(con.name, con.command as string[])
-                    })
+                if (Array.isArray(result.ephemeralContainers)) {
                     let ephemeralContainers = []
-                    result.manifest.status.ephemeralContainerStatuses?.forEach((_container) => {
-                        //con.state contains three states running,waiting and terminated
-                        // at any point of time only one state will be there
-                        if (_container.state.running) {
-                            ephemeralContainers.push({
-                                name: _container.name,
-                                isInitContainer: false,
-                                isEphemeralContainer: true,
-                                isExternal: isExternalEphemeralContainer(
-                                    ephemeralContainerStatusMap.get(_container.name),
-                                    _container.name,
-                                ),
-                            })
-                        }
+                    result.ephemeralContainers?.forEach((_container) => {
+                        ephemeralContainers.push({
+                            name: _container.name,
+                            isInitContainer: false,
+                            isEphemeralContainer: true,
+                            isExternal: _container.isExternal,
+                        })
                     })
```

Comment:
Suggestion: We can use result.ephemeralContainers.map(...) for a cleaner code

Reply:
done